osu.Game/Online/Chat/ColouredDrawableLinkCompiler.cs

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Collections.Generic;
+using osu.Framework.Allocation;
+using osu.Framework.Graphics;
+using osu.Game.Overlays;
+using osuTK.Graphics;
+
+namespace osu.Game.Online.Chat
+{
+    public class ColouredDrawableLinkCompiler : DrawableLinkCompiler
+    {
+        private Color4 idleColour;
+
+        public new Color4 IdleColour
+        {
+            get => idleColour;
+            set => base.IdleColour = idleColour = value;
+        }
+
+        private Color4 hoverColour;
+
+        public new Color4 HoverColour
+        {
+            get => hoverColour;
+            set => base.HoverColour = hoverColour = value;
+        }
```

Comment:
Not too sure about the whole colour setting logic - I didn't want to make these fields from `OsuHoverContainer` public, so I decided to overwrite them this way. Is there a better way to achieve what we need here?